dask/dataframe/core.py

```diff
@@ -3227,9 +3227,34 @@ def pivot_table(self, index=None, columns=None,
         return pivot_table(self, index=index, columns=columns, values=values,
                            aggfunc=aggfunc)
 
-    def to_records(self, index=False):
+    def to_records(self, lengths=None, index=False):
         from .io import to_records
-        return to_records(self)
+        from dask.array.core import normalize_chunks
+
+        if lengths is True:
+            lengths = tuple(self.map_partitions(len).compute())
+
+        records = to_records(self)
+
+        if isinstance(lengths, Sequence):
```

Comment:
Move this to a `_validate_chunks` method and call like

```
chunks = self._validate_chunks(records, lengths)
records._chunks = chunks
```

When `lengths` is None, you'll return `records._chunks` from `_validat_chunks`.

and call from `to_dask_array` as well.

Reply:
Would we need to test the new `_validate_chunks()` method?